mkosi/__init__.py

```diff
@@ -293,17 +293,20 @@ def configure_extension_release(context: Context) -> None:
         if "VERSION_ID" not in extrelease and (version := osrelease.get("VERSION_ID")):
             f.write(f"VERSION_ID={version}\n")
 
-        if f"{prefix}_ID" not in extrelease and context.config.image_id:
-            f.write(f"{prefix}_ID={context.config.image_id}\n")
+        if "ARCHITECTURE" not in extrelease:
+            f.write(f"ARCHITECTURE={context.config.architecture}\n")
 
-        if f"{prefix}_VERSION_ID" not in extrelease and context.config.image_version:
-            f.write(f"{prefix}_VERSION_ID={context.config.image_version}\n")
+        if f"{prefix}_ID" not in extrelease:
+            if id := context.config.environment.get(f"{prefix}_ID", context.config.image_id):
+                f.write(f"{prefix}_ID={id}\n")
 
-        if f"{prefix}_SCOPE" not in extrelease:
-            f.write(f"{prefix}_SCOPE=initrd system portable\n")
+        if f"{prefix}_VERSION_ID" not in extrelease:
+            if version_id := context.config.environment.get(f"{prefix}_VERSION_ID", context.config.image_version):
+                f.write(f"{prefix}_VERSION_ID={version_id}\n")
 
-        if "ARCHITECTURE" not in extrelease:
-            f.write(f"ARCHITECTURE={context.config.architecture}\n")
+        if f"{prefix}_SCOPE" not in extrelease:
+            if scope := context.config.environment.get(f"{prefix}_VERSION_ID"):
+                f.write(f"{prefix}_SCOPE={scope}")
```

Comment:
Copy paste bug? Also we should keep the existing default.

Reply:
Indeed was a copy-paste error. I do think having it by default use "upstream default" in this case ("system portable") does make more sense unless explicitly a different value is wanted.

But if you are adament on wanting to keep the existing default I can do that.